arviz/plots/kdeplot.py

```diff
@@ -229,8 +230,19 @@ def plot_kde(
             fill_func(fill_x, fill_y, **fill_kwargs)
 
         if legend and label:
-            legend_element = [Patch(edgecolor=default_color, label=label)]
-            ax.legend(handles=legend_element)
+            if fill_kwargs["alpha"] != 0:
+                legend_element = [
+                    Patch(
+                        facecolor=fill_kwargs["color"],
+                        edgecolor=plot_kwargs["color"],
+                        label=label,
+                        alpha=fill_kwargs["alpha"],
+                        linestyle=plot_kwargs["linestyle"],
```

Comment:
I would try to merge `fill_kwargs` and `plot_kwargs`, after removing some of the arguments like the color instead of passing them all "by hand". Other arguments like linewidth or hatch may also be relevant.

Reply:
Even though, it looks like from all the possible arguments of a Patch only these two I pointed out are relevant for a legend.

Reply:
I would merge both kwargs dicts, something like this should work:

    plot_options = ("ls", "linestyle", "lw", "linewidth")
    fill_options = ("alpha", "hatch", "capstyle", "joinstyle")
    kwargs1 = {key: value for key, value in fill_kwargs.items() if key in fill_options} 
    kwargs2 = {key: value for key, value in plot_kwargs.items() if key in plot_options}
    patch_kwargs = {**kwargs1, **kwargs2}

And then, when calling patch, pass the 2 colors and the label manually, and then `**patch_kwargs`.